substrate/frame/broker/src/dispatchable_impls.rs

```diff
@@ -437,4 +436,25 @@ impl<T: Config> Pallet<T> {
 		Self::deposit_event(Event::AllowedRenewalDropped { core, when });
 		Ok(())
 	}
+
+	pub(crate) fn ensure_core_availability(
+		status: &StatusRecord,
+		sale: &SaleInfoRecordOf<T>,
+	) -> Result<(), DispatchError> {
+		ensure!(sale.first_core < status.core_count, Error::<T>::Unavailable);
+		ensure!(sale.cores_sold < sale.cores_offered, Error::<T>::SoldOut);
```

Comment:
What is the motivation to split this into another function?

Reply:
To avoid code duplication: https://github.com/paritytech/polkadot-sdk/pull/3485#discussion_r1537393976

But, yeah, I don't have a strong opinion here since it's only two lines anyway.

Reply:
Doh, I read that comment and completely missed it. I can see both sides, if something else wants to use it then it makes sense to split it out.